schematic/store/synapse.py

```diff
@@ -51,9 +52,102 @@
 from schematic.exceptions import MissingConfigValueError, AccessCredentialsError
 
 from schematic import CONFIG
+from schematic.utils.general import entity_type_mapping
 
 logger = logging.getLogger("Synapse storage")
 
+@dataclass
+class ManifestDownload(object):
+    """
+    syn: an object of type synapseclient.
+    manifest_id: id of a manifest  
+    """
+    syn: synapseclient.Synapse
+    manifest_id: str
+
+    def _download_manifest_to_folder(self):
+        """
+        try downloading a manifest to local cache or a given folder
+        manifest
+        Return: 
+            manifest_data: A new Synapse Entity object of the appropriate type
+        """
+         # TO DO: potentially deprecate the if else statement because "manifest_folder" key always exist in config
+        if CONFIG["synapse"]["manifest_folder"]:
+            manifest_data = self.syn.get(
+                    self.manifest_id,
+                    downloadLocation=CONFIG["synapse"]["manifest_folder"],
+                    ifcollision="overwrite.local",
+                )
+        else:
+            manifest_data = self.syn.get(
+                        self.manifest_id,
+                    )
+        return manifest_data 
+
+    def _entity_type_checking(self):
+        """
+        check the entity type of the id that needs to be downloaded
+        Return: 
+             if the entity type is wrong, raise an error
+        """
+        # check the type of entity
+        entity_type = entity_type_mapping(self.syn, self.manifest_id)
+        if entity_type  != "file":
+            logger.error(f'You are using entity type: {entity_type}. Please try using a file')
+
+    @staticmethod
+    def download_manifest(self, newManifestName="", manifest_df=None):
```

Comment:
nit: Try your best in getting in the habit of adding type hints. It's extremely painful to code to best practices especially when you're busy, but it will become second nature to you the more you do it.

Reply:
@thomasyu888 We are in the process of getting Schematic up to speed on the most up to date coding practices and this includes being fully typed and enforced with mypy. Its a pretty big codebase so it might take a while, but I agree it's very important. I know having schematic_db being fully typed has saved me a lot of possible bugs.

Reply:
Totally agree - it should be iterative. The intent here was to slowly guide new contributions or parts of new contributions to follow best practices so we don't add to the backlog of code that needs to be linted / refactored.

Reply:
Thanks Tom for catching all these. It has been a great learning experience for me.